pkg/cmd/admin/router/router.go

```diff
@@ -477,7 +479,8 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg *
 			return fmt.Errorf("router could not be created; you must specify a service account with --service-account")
 		}
 
-		err := validateServiceAccount(kClient, namespace, cfg.ServiceAccount)
+		var err error
+		accountExists, needsSCC, err = validateServiceAccount(kClient, namespace, cfg.ServiceAccount, cfg.HostNetwork)
```

Comment:
I don't see it - yes, they are defined on line number 472 within the scope of the function `RunCmdRouter` and then are *only* used within the block from line 473 to 651 (the `if generate` block of code). The other instances of those variables are from another function `validateServiceAccount` - really return values from that function.